server.go

```diff
@@ -73,3 +67,17 @@ func Listen(svc Service, addr string) (Server, error) {
 	}
 	return Serve(svc, l)
 }
+
+// ServeHTTPServer returns a typhon server for the given HTTP server.
+// This supports callers that that need to alter the configuration of
+// the HTTP server (e.g. to support TLS).
+func ServeHTTPServer(s *http.Server, l net.Listener) (Server, error) {
```

Comment:
@obeattie  Should I:
 - Name this `ServeHttpServer` to match `HttpServer`
 - Rename `HttpServer` to `HTTPServer` (technically a breaking change)
 - Name this `ServeHTTPServer` even though it is inconsistent